VotingApplication/VotingApplication.Web.Tests/E2E/BasicPollTests.cs

```diff
@@ -377,137 +256,146 @@ public static Poll CreatePoll(TestVotingContext testContext)
         }
 
         [TestClass]
-        public class NamedVotersPollConfiguration
+        public class NamedVotingTests : E2ETest
         {
-            private static ITestVotingContext _context;
-            private static Poll _namedBasicPoll;
+            const string VoterName = "User";
             private static readonly Guid PollGuid = Guid.NewGuid();
-            private static readonly string PollUrl = SiteBaseUri + "Poll/#/Vote/" + PollGuid;
-            private IWebDriver _driver;
+            private static readonly string PollVoteUrl = SiteBaseUri + "Poll/#/" + PollGuid + "/Vote";
+            private static readonly string PollResultsUrl = SiteBaseUri + "Poll/#/" + PollGuid + "/Results";
 
-            [ClassInitialize]
-            public static void ClassInitialise(TestContext testContext)
+            [TestMethod]
+            [TestCategory("E2E")]
+            public void NoNameEntered_VoteNotAllowed()
             {
-                _context = new TestVotingContext();
+                using (IWebDriver driver = Driver)
+                {
+                    using (var context = new TestVotingContext())
+                    {
+                        CreateNamedVotersPoll(context);
+                        GoToUrl(driver, PollVoteUrl);
 
-                List<Choice> testPollChoices = new List<Choice>() {
-                new Choice(){ Name = "Test Choice 1", Description = "Test Description 1" },
-                new Choice(){ Name = "Test Choice 2", Description = "Test Description 2" }};
 
-                // Open, Named voters, No Choice Adding, Shown Results
-                _namedBasicPoll = new Poll()
-                {
-                    UUID = PollGuid,
-                    PollType = PollType.Basic,
-                    Name = "Test Poll",
-                    LastUpdatedUtc = DateTime.UtcNow,
-                    CreatedDateUtc = DateTime.UtcNow,
-                    Choices = testPollChoices,
-                    InviteOnly = false,
-                    NamedVoting = true,
-                    ChoiceAdding = false,
-                    ElectionMode = false
-                };
+                        IWebElement voteButton = FindElementById(driver, "vote-button");
+                        voteButton.Click();
 
-                _context.Polls.Add(_namedBasicPoll);
-                _context.SaveChanges();
+                        Assert.IsFalse(driver.Url.StartsWith(PollResultsUrl));
+                        Assert.IsTrue(driver.Url.StartsWith(PollVoteUrl));
+                    }
```

Comment:
I think either we need a placeholder for an error here, or the vote button needs to not be clickable. (voteButton.Enabled == false)